### DISCUSSION TITLE
General UI improvements

### DISCUSSION THREAD

**Discussion by Larsklopstra** [filamentphp/filament#2236] [Ideas]
I wanted to share a small list of consistencies and improvements which should be addressed:

## Buttons

- ~~The `tracking-tighter` decreases legibility due to the usage of `font-medium`, I'd suggest to remove the tracking and keep the font-weight.~~
- I'd make the copy a little longer, so instead of "Save" I'd recommend "Save changes", it makes the button size look less awkward and draws more attention over cancel (which is currently longer)
- ~~The button font sizes are out of proportion in comparison to the UI, I'd recommend using `text-sm` (this also applies to the "add to builder" button)~~

#2242 

## Builder

~~The builder is a great field for a variety for use cases, however, with larger amount of items it loses its clarity.~~

- ~~The "Add to X" button could be more subtile, perhaps just use the standard secondary button and left align it.~~
- ~~Make the plus icon less dominant by making the color gray (400/500, depends on what's consistent with the UI)~~
- ~~Perhaps take some inspiration from Statamic's brilliant Bard fieldtype, they seem to use a gray background with card what even large space between them (card `1rem gap` gray plus button `1rem gap` card) [Here's what it looks like](https://statamic.dev/fieldtypes/replicator)~~

Being fixed in #2239 

## Dropdowns

The dropdowns blend in with the background, that doesn't look too great. We can easily fix this by applying a `ring` instead of a border with some opacity.

**With `ring-1 ring-gray-900/10`:**

<img width="347" alt="With" src="https://user-images.githubusercontent.com/25669876/165523116-6b4adbc4-6b21-4cac-b009-38222fa43bac.png">

---

**Without `ring`:**

<img width="347" alt="Without" src="https://user-images.githubusercontent.com/25669876/165523251-f724e19a-a0bf-436a-af66-8032f0646c0a.png">

## Theming selectors

The current theme-ing selectors are great! However, having more wouldn't hurt :) My current suggestions:

- `filament-sidebar-item-link` actually styles the `a` instead of `li`
- `filament-sidebar-item-link-active` make it possible to style the active state

## Cards

~~Some cards are not uniform with other UI elements, I'd suggest applying `border border-gray-300 shadow-sm` to them. (eg: widget cards are missing the borders.)~~

#2241 

## Tables

The primary color usage on hover is a bit off-putting to me. It doesn't blend very well with the row borders and text. Just keep it simple and use `bg-gray-50`

**Before:**

<img width="900" alt="Before" src="https://user-images.githubusercontent.com/25669876/165524523-cfe2ceff-250a-4ffd-9c8a-d188b19f015c.png">

---

**After:**

<img width="900" alt="image" src="https://user-images.githubusercontent.com/25669876/165524642-0e83a136-6d15-4bf9-a15c-81781ddfae58.png">

## Closing

That's all what I can think off, it looks pretty great and consistent otherwise!

**Comment by Larsklopstra** [comment]
It'd like to add a few more nitpicks:

- Use a denser sidebar width (256px/16rem?)
- Don't constrain the topbar container width to 7xl, let it be fluid
- Add resource/page actions under a dropdown rather than a row of buttons

**Reply by ryangjchandler** [reply]
@Larsklopstra The row of action buttons is the default, me and Dan have already spoken about adding an `ActionDropdown` class that would wrap them all. I started working on a PoC a little while ago but haven't finished it yet.

**Reply by danharrin** [reply]
`ActionGroup` has been implemented now

I don't think that the sidebar width should change as this would be a breaking change, but it's now using a CSS variable, so you can customize that

**Comment by danharrin** [comment]
RE: suggested sidebar CSS classes

`.filament-sidebar-item-active` has been implemented, `-link` could be done easily with `.filament-sidebar-item a`

**Comment by danharrin** [comment]
RE: dropdown border being replaced with a ring

#3443

**Comment by danharrin** [comment]
Row hover color was fixed a few months ago.

**Comment by danharrin** [comment]
Apart from my other comments, all the rest of the feedback has been implemented in #3602. Thank you very much @Larsklopstra! Additional feedback is always welcome.
